app/scripts/controllers/network-set-controller.js

```diff
@@ -287,6 +340,8 @@ ndexApp.controller('networkSetController',
                 "indexLevel"    :   indexLevel
             };
             $scope.networkGridOptions.data.push(row);
+            $scope.renderNetworkTable = true;
+            setTimeout($scope.networkGridApi.core.handleWindowResize, 250);
```

Comment:
Why do we need a timeout now? Why is 250 chosen?

Reply:
Time triggers table refresh event.  This patch helps to fix the UI Grid issue where table gets distorted over slow connection.

Reply:
250 millisecs is enough time to re-draw the table.

Reply:
I am confused here. Your first comment says 250 ms is for slow connection, do you mean you expect the all data receive within 250 ms? In your 2nd comment, 250 is for re-drawing the table. Does that mean it is not depending on the data? What will happen if it takes more than 250 ms to receive all the data from server?

Reply:
WE DO NOT SET TIMERS TO RECEIVE DATA!!!!!!!
THE TIMERS HAVE NOTHING TO DO WITH RECEIVING DATA!!!!!!!!!
WHEN TIMER IS SET ALL DATA HAVE ALREADY BEEN RECEIVED FROM THE SERVER!!!!!!!

Timer is used to 'patch" the internal bug in UI Grid when table is distorted ....
Let me stop by and give you a small demo.

Reply:
WOW!!!!!!!! THIS IS A NEW CONVENTION WE ARE USING TO COMMUNICATE TECHNICAL ISSUES ????????